### PR TITLE
Refactor nodeManager instantiation logic

### PR DESCRIPTION
**Pull request by aleksanb**
Nonexisting constructors (due to race conditions) and errors in your
node's constructor are now two separate errors again, to everyones
delight!

### CODE REVIEW

File: nin/frontend/app/scripts/controllers/main.js [hunk 1]
```diff
@@ -78,7 +78,8 @@ class MainCtrl {
             try {
               let node = demo.nm.createNode(nodeInfo);
               demo.nm.insertOrReplaceNode(node);
-            } catch (e) {
+            } catch(e) {
+              console.error(e);
```
Comment (sigvef) [inline nin/frontend/app/scripts/controllers/main.js]: Will this flood the logs with the same error 5 times a second? Should we differentiate between "node hasn't been loaded yet" vs "node crashed when we tried to init it"?

Comment (aleksanb) [inline nin/frontend/app/scripts/controllers/main.js]: We should, yes. Currently the code is a spaghetti ball of yarn, and needs a refactor to be able to properly differentiate between the two cases.

**Comment by aleksanb** [comment]
Updated to comply with the NIN2017 release.

File: nin/dasBoot/NodeManager.js [hunk 2]
```diff
@@ -7,19 +7,16 @@ class NodeManager {
...
-        node = new NIN[nodeInfo.type.slice(4)](nodeInfo.id, nodeInfo.options);
-      } else {
-        node = new window[nodeInfo.type](nodeInfo.id, nodeInfo.options);
-      }
-    } catch (e) {
-      const errorMessage = "NodeManager: Attempted to instantiate nonexisting node type: " + nodeInfo.type;
-      console.error(errorMessage);
-      throw new Error(errorMessage);
+
+    const constructor = nodeInfo.type.slice(0, 4) === 'NIN.' ?
```
Comment (stianjensen) [inline nin/dasBoot/NodeManager.js]: Using constructor as a variable name _might_ work, but it also has meaning in js, so could we use something else here?

Comment (aleksanb) [inline nin/dasBoot/NodeManager.js]: How about `const radicalCat = ...` ?

Comment (aleksanb) [inline nin/dasBoot/NodeManager.js]: `const utroTorsk = ...`?

Comment (stianjensen) [inline nin/dasBoot/NodeManager.js]: I was thinking something more in the lines of `nodeConstructor` 🤔

Comment (aleksanb) [inline nin/dasBoot/NodeManager.js]: `const nodeConstructor = ...`?

**Comment by aleksanb** [comment]
Ping.

**Pull request merged**